rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+
+2. Determine Input UTXOs and Public State Modifications: The sender gathers all the input UTXOs needed for the transaction and determines the public state modifications 
+necessary for the Nescience applications and token transfers involved.
+
+3. Calculate Nullifiers: Nullifiers are generated for each input UTXO, ensuring that these UTXOs cannot be reused or double-spent. The nullifiers are derived from the
+ corresponding UTXO commitments.
+
+4. Call zkVM with Deshielded Circuits: The sender invokes the zkVM with deshielded kernel circuits, which generates the proof. The proof ensures that all input UTXOs 
+are valid by verifying their membership in the UTXO tree and their non-membership in the nullifier tree, ensuring they haven’t been spent.
+
+5. Generate and Submit Proof: The zkVM generates a zero-knowledge proof that verifies the correctness of the transaction without revealing private details. 
+The proof includes the nullifiers and the planned modifications to the public state.
+
+6. Send Proof to Sequencer: The sender then sends the proof and any relevant public information to the sequencer. The sequencer is responsible for verifying the proof, 
+updating the public state accordingly, and adding the nullifiers to the nullifier tree.
+
+Once the proof and public information have been broadcast to the network, the receiver does not need to take any further action. 
+The sequencer manages the public state updates and ensures that the transaction is properly executed. By the end of the deshielded execution, 
+specific transaction details become publicly visible, such as the identity of the receiver and the outcome of the transaction. 
+This allows participants in the public state to extract information about the transaction, including the receiver's identity and some details about the execution. 
+While the receiver's identity is revealed, the sender's identity and sensitive transaction details remain hidden, thanks to the use of zero-knowledge proofs. 
+This makes deshielded executions ideal for cases where transparency is needed, but complete privacy is still a priority for certain elements of the transaction.
+
+### **Summary of UTXOs Consumption in the Nescience Architecture**
+
+In the Nescience architecture, consuming UTXOs is a critical step in maintaining the security and integrity of the blockchain by preventing double-spending. 
+When a UTXO is consumed, it is used as an input in a transaction, effectively marking it as spent. This ensures that the UTXO cannot be reused, preserving the integrity of the blockchain.
+
+1. The Process of Consuming UTXOs: The process of consuming a UTXO begins when a user selects a UTXO from their private state. The user verifies the UTXO’s existence and 
+ownership using their viewing key, ensuring that they are the legitimate owner of the UTXO. Once verified, the user generates two key cryptographic proofs:
```

Comment:
This is a little ambiguous. Are we using key as in "critical", "essential" or "important" or as in "public and private key"